server/src/test/java/com/octopus/teamcity/opentelemetry/server/OTELHelperTest.java

```diff
@@ -0,0 +1,84 @@
+package com.octopus.teamcity.opentelemetry.server;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import jetbrains.buildServer.serverSide.SRunningBuild;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class OTELHelperTest {
+
+    @Mock private OpenTelemetry openTelemetry = mock(OpenTelemetry.class, RETURNS_DEEP_STUBS);
+    @Mock private Tracer tracer = mock(Tracer.class, RETURNS_DEEP_STUBS);
+    @Mock private HashMap<String, Span> spanMap;
+    private OTELHelper otelHelper;
+
+    @BeforeEach
+    void setUp() {
+        final String ENDPOINT = "https://otel.endpoint.com";
+        final Map<String, String> HEADERS = new HashMap<>() {{
+            put("testHeader1", "testHeaderValue1");
+            put("testHeader2", "testHeaderValue2");
+        }};
+        GlobalOpenTelemetry.resetForTest();
+        this.otelHelper = new OTELHelper(HEADERS, ENDPOINT);
+    }
+
+
+    @Test
+    void isReadyTest() {
+        assertNotNull(this.openTelemetry);
+        assertNotNull(this.tracer);
+        assertNotNull(this.spanMap);
+    }
+
+    @Test
+    void getParentSpanTest(@Mock SRunningBuild build) {
+        assertEquals(this.otelHelper.getParentSpan(String.valueOf(build.getBuildId())), this.otelHelper.getSpan(String.valueOf(build.getBuildId())) );
```

Comment:
Consider splitting this into procedural steps and maybe add commentary that the call to `getParentSpan` is expected to create data and that `getSpan` is expected to return itself as parent when nothing else is supplied